src/Migrations/M028MovePageHeaderSideBarOptions.php

```diff
@@ -0,0 +1,222 @@
+                            >
+                                ' . $page_header_button_title . '
+                            </a>
+                        </div>
+                    <!-- /wp:buttons -->'
+                : '') . '
+
+                </div>
+                <figure class="wp-block-media-text__media">
+                <img
+                    src=' . $page_header_image . '
+                    alt=""
+                    class="wp-image-' . $page_header_image_id . ' size-full"
+                />
+                </figure></div>
+                <!-- /wp:media-text --></div>
+                <!-- /wp:group --></div>
+                <!-- /wp:group -->
+                <!-- /wp:planet4-block-templates/page-header -->
+                ';
+
+                $updated_post_content = $content . $page->post_content;
+
+                $keys_to_update = array(
+                    'p4_title',
+                    'p4_subtitle',
+                    'background_image',
+                    'background_image_id',
+                    'p4_description',
+                    'p4_button_title',
+                    'p4_button_link',
+                    'p4_button_link_checkbox',
+                    'p4_hide_page_title_checkbox',
+                );
+
+                // Update specific meta values to empty strings
+                foreach ($keys_to_update as $meta_key) {
+                    if ($meta_key === 'p4_hide_page_title_checkbox' && !empty($page_header_title)) {
+                        if (isset($page_meta_data[$meta_key]) && in_array('on', $page_meta_data[$meta_key])) {
+                            continue;
+                        }
+
+                        // Update meta value with string 'on'
+                        update_post_meta($page->ID, $meta_key, 'on');
+                    } else {
+                        update_post_meta($page->ID, $meta_key, '');
+                    }
+                }
+
+
+                $post_args = array(
+                    'post_content' => $updated_post_content,
+                    'ID' => $page->ID,
+                );
+
+                wp_update_post($post_args);
+
+                continue;
+            }
+
+            if (empty($page_header_image_id) || empty($page_header_title)) {
+                if (
+                    empty($page_header_image_id)
+                    && empty($page_header_title)
+                    && empty($page_header_description)
+                    && empty($page_header_button_title)
+                ) {
+                    continue;
+                }
+
+                $content = '
+                    <!-- wp:group {"layout":{"type":"flex","orientation":"vertical"}} -->
+                        ' . (!empty($page_header_title) ?
+                            '<div class="wp-block-group"><!-- wp:heading -->
+                                <h1 class="wp-block-heading">' . $page_header_title . '</h1>
```

Comment:
[default level](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/heading/block.json#L22) for wp-heading is `2`, if you want to use `h1` you have to specify it with `<!-- wp:heading {"level":1} -->`
Side question, do we want the class `page-header-title` for these titles to match the current layout ?

Reply:
I will replaces the class then so we have a consistent layout.